crates/hir/src/lib.rs

```diff
@@ -415,6 +415,19 @@ impl ModuleDef {
             def.diagnostics(db, &mut acc);
         }
 
+        let fields = match self {
+            ModuleDef::Adt(Adt::Struct(it)) => Some(it.fields(db)),
+            ModuleDef::Adt(Adt::Union(it)) => Some(it.fields(db)),
+            ModuleDef::Variant(it) => Some(it.fields(db)),
+            _ => None,
+        };
+        if let Some(fields) = fields {
+            for field in fields {
+                let def: DefWithBody = field.into();
+                def.diagnostics(db, &mut acc, style_lints);
```

Comment:
I do worry about the perf impact this PR might have wrt to memory (and revision refreshing due to so many new query entries) as now every field becomes a body, and no one will really be using this feature yet so basically all fields will be default-less. So I'd propose we e ither do not compute diagnostics for these at all for now, OR, give variants a flag for whether they contain any defaulted-fields or not as a small optimization that allows us to skip this if we know that none of their fields have any bodies.